### ISSUE TITLE
Wrong commands sequence. ShellStream.WriteLine didn't flush buffer before send its data. 

### ISSUE THREAD

**Issue by theanurin** [sshnet/SSH.NET#180]
```
using (var sshStream = sshClient.CreateShellStream("C#", 80, 25, 100, 100, 1000))
{
    new MemoryStream(sshClient.ConnectionInfo.Encoding.GetBytes("echo 'command 1'\r")).CopyTo(sshStream);
    sshStream.WriteLine("echo 'command 2'");
    sshStream.Flush();
}

```
Expected result:
   command 1
   command 2

Current result:
  command 2
  command 1

My use case:
 I open embedded resource (somescript.sh) as stream and want to send all content of the somescript.sh to sshStream...

**Comment by IgorMilavec** [comment]
I have created a discussion #926 for all ShellStream related issues.

**Comment by WojciechNagorski** [comment]
This issue has been fixed in the 2024.0.0 version.
